### PR TITLE
refactor Spinner, kill SVGSpinner

### PR DESCRIPTION
**Pull request by giladgray**
- refactor `Spinner` markup considerably:
  - remove all wrapper elements so svg is the root
  - `overflow:visible` means no need to switch viewBox anymore!
  - animate only the track head path, not the whole thing
- 🔥  delete `SVGSpinner` since `Spinner` can now be used inside an SVG (no DOM elements)
- 🔥  delete `Classes.SPINNER_SVG_CONTAINER` and `Classes.SVG_SPINNER`

### CODE REVIEW

File: packages/core/src/components/spinner/_spinner.scss [hunk 1]
```diff
@@ -29,18 +27,21 @@ Styleguide spinner
...
 }
 
-// see https://css-tricks.com/scale-svg/
-// how to Scale SVG to Fit the Available Width (and adjust the height to match) Option 4
-
 .#{$ns}-spinner {
   @include spinner-size($spinner-width-factor, $spinner-stroke-width, $spinner-speed);
+  // allow paths to overflow container -- critical for edges of circles!
+  overflow: visible;
+  vertical-align: top;
```
Comment (themadcreator) [inline packages/core/src/components/spinner/_spinner.scss]: is vertical align necessary? is this going to mess with existing uses?

Comment (giladgray) [reply packages/core/src/components/spinner/_spinner.scss]: hmm good thought. i'll change to `middle`

**Comment by giladgray** [comment]
need to add `height` property to ensure spinner is square

File: packages/core/src/common/classes.ts [hunk 2]
```diff
@@ -193,8 +193,6 @@ export const SPINNER = `${NS}-spinner`;
 export const SPINNER_HEAD = `${SPINNER}-head`;
 export const SPINNER_NO_SPIN = `${NS}-no-spin`;
```
Comment (llorca) [inline packages/core/src/common/classes.ts]: while we're at it, thoughts on a better name for this? something more generic like `.pt-static` maybe? or maybe indeterminate vs determinate loading?

File: packages/core/src/components/spinner/spinner.tsx [hunk 3]
```diff
@@ -50,36 +50,18 @@ export class Spinner extends React.PureComponent<ISpinnerProps, {}> {
...
-        );
-    }
-
-    // abstract away the container elements so SVGSpinner can do its own thing
-    protected renderContainer(classes: string, content: JSX.Element) {
         return (
-            <div className={classes}>
-                <div className={Classes.SPINNER_SVG_CONTAINER}>{content}</div>
-            </div>
+            <svg className={classes} viewBox="0 0 100 100">
```
Comment (llorca) [inline packages/core/src/components/spinner/spinner.tsx]: I don't fully understand SVG/viewBox and stuff, but it's amazing that you were able to dramatically simplify

File: packages/core/src/components/spinner/_spinner.scss (hunk 1 above)
Comment (llorca) [reply packages/core/src/components/spinner/_spinner.scss]: is it even needed?

**Pull request merged**